File: solid/max_miguel/typescript/EmpleadoTiempoCompleto.ts
import Empleado from "./Empleado";

export default class EmpleadoTiempoCompleto extends Empleado {
  private sueldoFijo: number;

  constructor(sueldoFijo: number, nombre: string, id: number) {
    super(nombre, id);
    this.sueldoFijo = sueldoFijo;
  }

  public calcularSueldo(): number {
    return this.sueldoFijo;
  }
}

File: solid/max_miguel/python/empleado_tiempo_completo.py
from empleado import Empleado

class EmpleadoTiempoCompleto(Empleado):
    def __init__(self, nombre, id, salario_mensual):
        super().__init__(nombre, id)
        self._salario_mensual = salario_mensual

    def calcular_salario(self):
        return self._salario_mensual